### DISCUSSION TITLE
Are container ID's guaranteed to be unique across users in rootless podman?

### DISCUSSION THREAD

**Discussion by cjvolzka** [containers/podman#24533] [Q&A]
The title sums it up. Are container ID's guaranteed to be unique across users in rootless podman?

According to the [OCI Runtime spec](https://github.com/opencontainers/runtime-spec/blob/main/runtime.md#state)
```
id (string, REQUIRED) is the container's ID. This MUST be unique across all containers on this host. There is no requirement that it be unique across hosts.
```
Since I don't believe there's a central way to list all containers across users in rootless podman, I want to to confirm if running container IDs are still guaranteed to always be unique when running rootless podman.

**Comment by rhatdan** [comment]
I don't believe so, and I don't think it matters as long as they are unique for each user session.  Container IDs are per instance of the podman database.

**Reply by Luap99** [reply]
Right there is guarantee as podman can only check its own/db storage so other user cannot be checked.

However I want to highlight that ids are created from random 32 bytes encoded as 64 char hex string. So we have 2^256 ids. Getting a conflict there is close to impossible (assuming a proper random generator)
